### ISSUE TITLE
EarlyStoppingCallback does not work

### ISSUE THREAD

**Issue by DogitoErgoSum** [UKPLab/sentence-transformers#2911]
I've been trying to get the EarlyStoppingCallback to work, but without success.

[Here](https://github.com/UKPLab/sentence-transformers/issues/1494), it says that the callback works. However, I haven't been able to get it to function with either the validation loss or any of the evaluator metrics.

Can you provide a working example or clarify whether the EarlyStoppingCallback is functional or not?

Package versions:
sentence-transformers==3.0.1
transformers==4.41.2

**Comment by ir2718** [comment]
@DogitoErgoSum 

Hi,

I've tried it out with the [STSB script](https://github.com/UKPLab/sentence-transformers/blob/master/examples/training/sts/training_stsbenchmark.py), and the same dependency versions. Here's the code I had to modify:

```

# 5. Define the training arguments
args = SentenceTransformerTrainingArguments(
    # Required parameter:
    output_dir=output_dir,
    # Optional training parameters:
    num_train_epochs=num_epochs,
    per_device_train_batch_size=train_batch_size,
    per_device_eval_batch_size=train_batch_size,
    warmup_ratio=0.1,
    fp16=True,  # Set to False if you get an error that your GPU can't run on FP16
    bf16=False,  # Set to True if you have a GPU that supports BF16
    # Optional tracking/debugging parameters:
    evaluation_strategy="steps",
    eval_steps=100,
    save_strategy="steps",
    save_steps=100,
    save_total_limit=2,
    logging_steps=100,
    run_name="sts",  # Will be used in W&B if `wandb` is installed
    load_best_model_at_end=True
)

# 6. Create the trainer & start training
trainer = SentenceTransformerTrainer(
    model=model,
    args=args,
    train_dataset=train_dataset,
    eval_dataset=eval_dataset,
    loss=train_loss,
    evaluator=dev_evaluator,
    callbacks=[
        EarlyStoppingCallback(
            early_stopping_patience=1,
            early_stopping_threshold=0
        )
    ]
)
```
Hope this helps, otherwise please share your code.

**Comment by DogitoErgoSum** [comment]
Thank you!
